src/components/RosterForLoop/RosterForLoop.tsx

```diff
@@ -60,22 +64,50 @@ export const RosterForLoop = (
 		handleChanges(newResponses);
 	}, [nbRows, min, valueMap, handleChanges]);
 
+	const removeRowWithIndex = useCallback(
+		(indexToRemove: number) => {
+			if (nbRows <= min) {
+				return;
+			}
+			/**
+			 * Case 0: trying to delete
+			 */
+			if (indexToRemove >= nbRows || indexToRemove < 0) {
+				return;
+			}
+			const newResponses = Object.entries(valueMap).map(([k, v]) => {
+				return {
+					name: k,
+					value: v?.filter((_, i) => i !== indexToRemove),
+					removedIndex: indexToRemove,
+				};
+			});
+			handleChanges(newResponses);
+			setNbRows((n) => n - 1);
+		},
+		[nbRows, min, valueMap, handleChanges]
+	);
+
 	if (nbRows === 0) {
 		return null;
 	}
 
 	let cols = 0;
 
+	const headerWithActions = header
+		? [...header, { label: D.ACTION_HEADER }]
+		: undefined;
+
 	return (
 		<CustomLoop
 			{...props}
 			errors={getComponentErrors(errors, props.id)}
 			addRow={nbRows === max ? undefined : addRow}
-			removeRow={nbRows === min ? undefined : removeRow}
+			removeRow={cantRemove ? undefined : removeRow}
 			canControlRows={!!(min && max && min !== max)}
 		>
 			<Table id={id}>
-				{header && <TableHeader header={header} />}
+				{headerWithActions && <TableHeader header={headerWithActions} />}
```

Comment:
Why not keep  it simple since `headerWithActions` is only used here?

```jsx
{header && <TableHeader header={[...header, { label: D.ACTION_HEADER }]} />}
```